File: dp/tabulation/fib.js
const fib = (n) => {
  if (n < 0) return n;
  const table = Array(n + 1).fill(0);
  table[1] = 1;
  for (let i = 0; i <= n; i++) {
    console.log(table[i + 2], "i + 2");
    table[i + 1] += table[i];
    table[i + 2] += table[i];
  }
  return table[n];
};

console.log(fib(13));

File: dp/tabulation/fib.py
def fib(n):
    if n < 0:
        return n
    table = [0] * (n + 1)
    table[1] = 1

    for i in range(n + 1):
        if i <= n - 2:
            table[i + 2] += table[i]
        if i <= n - 1:
            table[i + 1] += table[i]
    return table[n]


print(fib(500000))
